### PR TITLE
Data Catalog tag template

### PR DESCRIPTION
**Pull request by danawillow**
Part  of terraform-providers/terraform-provider-google#5892

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`data_catalog_tag_template`
```

### CODE REVIEW

**Comment by danawillow** [comment]
Open questions:
* Right now I've set `force=true` for delete all the time, but should it be its own field so people set it explicitly?
* TF doesn't have support for string -> object maps, which is why fields are implemented as a set of objects (with a "key" field of field_id). Will the tag resources need to reference anything in here? If so we may need to think about how they would do that, since elements of sets aren't easily referenceable

**Comment by megan07** [comment]
> Open questions:
> 
> * Right now I've set `force=true` for delete all the time, but should it be its own field so people set it explicitly?
> * TF doesn't have support for string -> object maps, which is why fields are implemented as a set of objects (with a "key" field of field_id). Will the tag resources need to reference anything in here? If so we may need to think about how they would do that, since elements of sets aren't easily referenceable

Sorry, I missed this.
* From my understanding, `force=true` was currently the only option. Would adding it as a field in the future be considered a breaking change, as long as we default to `true`? If it'd be considered breaking, then maybe we should add it now and document it in the same was as the API has it documented
* I've got the tags working with this set up so far, so I think we're good - if you see anything while reviewing the Tag resource, we might have to reconsider, but I think it will be fine :)

**Comment by danawillow** [comment]
`force=true` is the only option that lets it get deleted successfully; otherwise it errors. It's basically the inverse of a `deletion_protection` field (since it defaults to false). I'm fairly inclined to just add it now (maybe call it `force_delete`?) after all the bigtable hullabaloo.

**Comment by megan07** [comment]
Sure!  That sounds good to me!

File: products/datacatalog/terraform.yaml [hunk 1]
```diff
@@ -84,6 +84,11 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         primary_resource_id: "basic_tag_template"
         vars:
           tag_template_id: "my_template"
+          force_delete: "false"
```
Comment (megan07) [inline products/datacatalog/terraform.yaml:87]: this is for the docs, correct?  do we want to put `false` in there yet before it's supported by the API?  I don't have a strong opinion either way, just wanted to check thoughts.

Comment (danawillow) [reply products/datacatalog/terraform.yaml:87]: Yup, same rationale here as the bigtable one where I prefer to recommend the safer option for users and make them explicitly have to switch over to the one that lets them delete the resource when they decide to do so.

**Pull request merged**